cfgmgr/intfmgr.cpp

```diff
@@ -150,13 +270,59 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
         }
 
         // Set Interface VRF except for lo
+        // Set sub interface except for lo
         if (!is_lo)
         {
             if (!vrf_name.empty())
             {
                 setIntfVrf(alias, vrf_name);
             }
-            m_appIntfTableProducer.set(alias, data);
+
+            if (!subIntfAlias.empty())
+            {
+                if (m_subIntfList.find(subIntfAlias) == m_subIntfList.end())
+                {
+                    try
+                    {
+                        addHostSubIntf(alias, subIntfAlias, vlanId);
+                    }
+                    catch (const std::runtime_error &e)
+                    {
+                        SWSS_LOG_NOTICE("Sub interface ip link add failure. Runtime error: %s", e.what());
+                        return false;
+                    }
+
+                    m_subIntfList.insert(subIntfAlias);
+                }
+
+                if (mtu)
+                {
+                    try
+                    {
+                        setHostSubIntfMtu(subIntfAlias, mtu);
+                    }
+                    catch (const std::runtime_error &e)
+                    {
+                        SWSS_LOG_NOTICE("Sub interface ip link set mtu failure. Runtime error: %s", e.what());
+                        return false;
+                    }
+                }
+
+                try
+                {
+                    setHostSubIntfAdminStatus(subIntfAlias, adminStatus);
+                }
+                catch (const std::runtime_error &e)
+                {
+                    SWSS_LOG_NOTICE("Sub interface ip link set admin status %s failure. Runtime error: %s", adminStatus.c_str(), e.what());
+                    return false;
+                }
+
+                // set STATE_DB port state
+                setSubIntfStateOk(subIntfAlias);
+            }
+
+            m_appIntfTableProducer.set(subIntfAlias.empty() ? alias : subIntfAlias, data);
```

Comment:
This data may not have `mtu `or `admin status` but you set `admin status` to `up`. You may want to add this defaults to table for `orchagent`.

Reply:
admin_status up is set to the CONFIG_DB VLAN_SUB_INTERFACE table when generating keys to the table https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/minigraph.py#L774

Reply:
mtu is not configurable in the current design but inherited from the parent physical port. https://github.com/Azure/SONiC/blob/master/doc/subport/sonic-sub-port-intf-hld.md#21-configuration

So it is not set to the CONFIG_DB VLAN_SUB_INTERFACE table

Reply:
Added both, admin_status and mtu, to field value tuples set to APPL_DB